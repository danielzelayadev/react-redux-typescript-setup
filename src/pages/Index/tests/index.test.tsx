import { shallow } from 'enzyme';
import * as R from 'ramda';
import * as React from 'react';
import { mapDispatchToProps, mapStateToProps, UI as Index } from '..';

describe('Index Page UI', () => {
  it('should render correctly', () => {
    expect(
      shallow(<Index appClicks={0} addClick={R.always(0)} />)
    ).toMatchSnapshot();
  });
});

describe('Index Page Container', () => {
  it('should map state to props correctly', () => {
    expect(mapStateToProps({ ui: { clicks: 10 } })).toMatchSnapshot();
  });
  it('should map dispatch to props correctly', () => {
    expect(mapDispatchToProps).toMatchSnapshot();
  });
});
