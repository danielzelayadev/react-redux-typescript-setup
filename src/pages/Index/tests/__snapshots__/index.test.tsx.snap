// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page Container should map dispatch to props correctly 1`] = `
Object {
  "addClick": [Function],
}
`;

exports[`Index Page Container should map state to props correctly 1`] = `
Object {
  "appClicks": 10,
}
`;

exports[`Index Page UI should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UI
    addClick={[Function]}
    appClicks={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "background": "#fff",
      "children": Array [
        <h1>
          Welcome to Rapture
        </h1>,
        <p>
          Number of Clicks is 
          0
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to Rapture",
        },
        "ref": null,
        "rendered": "Welcome to Rapture",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Number of Clicks is ",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Number of Clicks is ",
          0,
        ],
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "background": "#fff",
        "children": Array [
          <h1>
            Welcome to Rapture
          </h1>,
          <p>
            Number of Clicks is 
            0
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Rapture",
          },
          "ref": null,
          "rendered": "Welcome to Rapture",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Number of Clicks is ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Number of Clicks is ",
            0,
          ],
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
